include/deal.II/base/types.h

```diff
@@ -78,13 +82,21 @@ namespace types
 
 #else
   /**
-   * The type used for global indices of degrees of freedom. While in
-   * sequential computations the 4 billion indices of 32-bit unsigned integers
-   * is plenty, parallel computations using the
+   * The type used to denote the global index of degree of freedom. This
+   * type is then also used for querying the global *number* of degrees
+   * of freedom, since the number is simply the largest index plus one.
+   *
+   * While in sequential computations the 4 billion indices of 32-bit unsigned
+   * integers is plenty, parallel computations using the
    * parallel::distributed::Triangulation class can overflow this number and
-   * we need a bigger index space.
+   * consequently, deal.II chooses a larger integer type when
+   * configured to use 64-bit indices.
+   *
+   * The data type always corresponds to an unsigned integer type.
    *
-   * The data type always indicates an unsigned integer type.
+   * See the
+   * @ref GlobalDoFIndex
+   * page for guidance on when this type should or should not be used.
```

Comment:
While at it, do we really need to duplicate the documentation or can we simply move the `#ifdef` around the variable only?